marlowe-playground-client/src/MainFrame.purs

```diff
@@ -65,8 +78,33 @@ mkMainFrame settings =
         }
     }
 
+toSimulation ::
+  forall m a.
+  Functor m =>
+  HalogenM ST.State ST.Action ChildSlots Message m a -> HalogenM FrontendState HAction ChildSlots Message m a
+toSimulation halogen = do
+  currentState <- get
+  let
+    setState = flip (set _simulationState) currentState
+  let
+    getState = view _simulationState
+  (imapState setState getState <<< mapAction SimulationAction) halogen
+
+toHaskellEditor ::
+  forall m a.
+  Functor m =>
+  HalogenM HE.State HE.Action ChildSlots Message m a -> HalogenM FrontendState HAction ChildSlots Message m a
+toHaskellEditor halogen = do
+  currentState <- get
+  let
+    setState = flip (set _haskellState) currentState
+  let
+    getState = view _haskellState
+  (imapState setState getState <<< mapAction HaskellAction) halogen
+
```

Comment:
I'm sure you could abstract this further to something like this, if you wanted:

```purescript
imapLens lens wrapper halogen = do
  currentState <- get
  let
    setState = flip (set lens) currentState
  let
    getState = view lens
  (imapState setState getState <<< mapAction wrapper) halogen
```

Reply:
cool, I'll see what I can do in the next PR